src/testsuite/arith128_test_f128.c

```diff
@@ -525,6 +522,674 @@ db_vec_xscvqpdpo (__binary128 f128)
+      //  2b) The exponent is == E_min
+      //
+      if (vec_cmpsd_all_lt ((vi64_t) q_exp, exp_min))
+	{
+	  print_vint128x (" tiny (q_exp):", (vui128_t) q_exp);
+	    {
+#if 1
+	      const vui64_t too_tiny = (vui64_t) CONST_VINT64_DW( 116, 116 );
+	      const vui32_t xmask = CONST_VINT128_W(0x1fffffff, -1, -1, -1);
+	      vui32_t tmp;
+#endif
+	      // Intermediate result is tiny, unbiased exponent < -16382
+	      //x_exp = vec_subudm ((vui64_t) exp_tiny, q_exp);
+	      x_exp = vec_subudm ((vui64_t) exp_min, q_exp);
+
+	      print_vint128x (" tiny (x_exp):", (vui128_t) x_exp);
+
+	      if  (vec_cmpud_all_gt ((vui64_t) x_exp, too_tiny))
+		{
+		  // Intermediate result is too tiny, the shift will
+		  // zero the fraction and the GR-bit leaving only the
+		  // Sticky bit. The X-bit needs to include all bits
+		  // from p_sig_h and p_sig_l
+		  p_sig_l = vec_srqi (p_sig_l, 8);
+		  p_sig_l = (vui128_t) vec_or ((vui32_t) p_sig_l, (vui32_t) p_sig_h);
+		  print_vint128x (" sig (h|l):", (vui128_t) p_sig_l);
+		  // generate a carry into bit-2 for any nonzero bits 3-127
+		  p_sig_l = vec_adduqm (p_sig_l, (vui128_t) xmask);
+		  q_sig = (vui128_t) q_zero;
+		  p_sig_l = (vui128_t) vec_andc ((vui32_t) p_sig_l, xmask);
+		  print_vint128x (" sig (pxl):", (vui128_t) p_sig_l);
+		}
+	      else
+		{ // Normal tiny, right shift may loose low order bits
+		  // from p_sig_l. So collect any 1-bits below GRX and
+		  // OR them into the X-bit, before the right shift.
+		  vui64_t l_exp;
+		  const vui64_t exp_128 = (vui64_t) CONST_VINT64_DW( 128, 128 );
+#if 1
+		  // Propagate low order bits into the sticky bit
+		  // GRX left adjusted in p_sig_l
+		  // Issolate bits below GDX (bits 3-128).
+		  tmp = vec_and ((vui32_t) p_sig_l, xmask);
+		  // generate a carry into bit-2 for any nonzero bits 3-127
+		  tmp = (vui32_t) vec_adduqm ((vui128_t) tmp, (vui128_t) xmask);
+		  // Or this with the X-bit to propagate any sticky bits into X
+		  p_sig_l = (vui128_t) vec_or ((vui32_t) p_sig_l, tmp);
+		  p_sig_l = (vui128_t) vec_andc ((vui32_t) p_sig_l, xmask);
+		  print_vint128x (" sig (pxl):", (vui128_t) p_sig_l);
+#endif
+
+		  l_exp = vec_subudm (exp_128, x_exp);
+
+		  print_vint128x (" tiny (l_exp):", (vui128_t) l_exp);
+		  p_sig_l = vec_sldq (p_sig_h, p_sig_l, (vui128_t) l_exp);
+		  p_sig_h = vec_srq (p_sig_h, (vui128_t) x_exp);
+		  print_vint128x (" srq (p_h):", (vui128_t) p_sig_h);
+		  print_vint128x (" srq (p_l):", (vui128_t) p_sig_l);
+		  q_sig = p_sig_h;
+		}
+	      q_exp = q_zero;
+	    }
+	}
+      else
+	{
+	  // Exponent is not tiny.
+	  if (vec_cmpuq_all_le (p_sig_h, (vui128_t) sigovt))
+	    {
+	      // But the significand is below normal range.
+	      // This can happen when multiplying a denormal by a
+	      // normal.
+	      const vui64_t exp_15 = { 15, 15 };
+	      vui64_t c_exp, d_exp;
+	      vui128_t c_sig;
+
+	      print_vint128x (" UFt (p_sig):", (vui128_t) p_sig_h);
+	      c_sig = vec_clzq (p_sig_h);
```

Comment:
consider merging initialization of `c_sig` with its definition 3 lines up.

Reply:
May change this later. Clear to me this way.